### DISCUSSION TITLE
In-band aliasing for resampler block?

### DISCUSSION THREAD

**Discussion by thezboe** [jgaeddert/liquid-dsp#338] [Q&A]
Looking at the `resamp_crcf_example.c` example, I'm not getting results that I expect from the resampler block. I'm under the impression that the resampler creates an anti-aliasing LPF (windowed sinc with firdes_kaiser), and then splits that up among a polyphase filter bank.  Changing nothing but the resampling rate from 1.1 to 0.75, the other defaults seem to be okay, I'm getting weird results.

<img width="357" alt="mstsc_iWvFxHXJu7" src="https://github.com/jgaeddert/liquid-dsp/assets/441183/943847ee-4323-4401-a616-af43a8b5b3db">

The tone is at 0.045, but I'm not expecting to see the massive images of the signal that are spaced 0.245 f/fs away. Even if I change the filter bandwidth to something very small, the images persist. I realize they are 60dB down, but I expect things that are present in the original signal to be attenuated by that much, not that it would create artifacts that aren't there.

Is this even aliasing? Wouldn't it have to be out-of-band for it to be considered that?

**Comment by jgaeddert** [comment]
Great question! The anti-aliasing filter in the example is designed with a 60 dB sidelobe suppression level. You'll notice that in your figure, the images are actually 60 dB down from the main signal. But this is only part of the story. The polyphase filterbank has only so much precision in time to give interpolants.

You can reduce the magnitude of these images by increasing the stop-band suppression level (e.g. to 80) and the number of filters in the bank (e.g. to 1024):

     ./examples/resamp_crcf_example -r 0.75  -p 1024 -s 80

<img width="614" alt="image" src="https://github.com/jgaeddert/liquid-dsp/assets/253027/fd5ebd71-2b7f-42ed-8a54-6648b9849f6d">

The "real" way to do this would be to interpolate between available sample points on the output of the filter bank; however this slows things down quite a lot (computation roughly triples), so I don't actually do this.
